### DISCUSSION TITLE
Theia behind HTTPD Apache proxy with sub path

### DISCUSSION THREAD

**Discussion by Kadalia** [eclipse-theia/theia#13530] [Q&A]
Hello,
I have theia running fine in docker container. http://localhost:3000.
I would like to access via https, so I set up an Apache proxy.
httpd conf looks like :

`<VirtualHost *:443>
    ServerName pm-05.XXX.com
    ServerAlias demo-jupiter.XXX.com
    ErrorLog /home/XXX/logs/httpd/error_log
    CustomLog /home/XXX/logs/httpd/access_log combined

    SSLEngine on
    SSLCertificateFile /home/XXX/conf/cert/pm-05.XXX.com.crt
    SSLCertificateKeyFile /home/XXX/conf/cert/pm-05.XXX.com.key

    ProxyRequests     Off

    RewriteEngine On
    RewriteCond %{HTTP:Upgrade} =websocket [NC]
    RewriteRule /theia/(.*)/     ws://localhost:3000/ [P,L]

    ProxyPass        /theia/ http://localhost:3000/
    ProxyPassReverse /theia/ http://localhost:3000/

</VirtualHost>
`
I can access theia with https://demo-jupiter.XXX.com/theia/
But I can't open any documents. Looks like a websocket issue.
Don't know how to make it right.
 
Thank's for your help !
Philippe

**Comment by msujew** [comment]
Hey @Kadalia,

do you get any errors in the browser devtools console? Without those, it's pretty difficult to figure out what's wrong.

**Reply by Kadalia** [reply]
Hello Mark,

Thank you for your comment.

I have no revelant error in the console. 
![image](https://github.com/eclipse-theia/theia/assets/7450654/4707eac9-e0af-4c23-8511-88ae0d2bb29b)
I can see a lot of socket.io requests with HTTP 200 return code. I am not familiar with web sockets, I don't know if it's normal.
As I am not familiar with Apache too ... I think that my httpd conf is totally wrong. I've tried a lot of conf ... this is the less "bad". 

I think that access Theia behind a proxy (to add SSL) with a sub path, not a sub domain should be quite easy .... but perhaps I'm wrong.
Do you have any hints or sample conf ?
I'm sorry, I can't switch to nginx or have a subdomain (certificates issue)

I've already set up a docker Theia behind an nginx with a sub domain, or a different IP port, works fine :-) but my question is more specific, I need to access docker Theia running in a VM via Apache to add SSL and I can't use a sub domain or another port than 443.

Philippe

**Reply by msujew** [reply]
Looks like the websocket connection isn't forwarded correctly and socket-io instead uses it's http long polling mode. But that in itself should work as expected.

@jfaltermeier maybe you know more?

**Reply by jfaltermeier** [reply]
I don't have much experience with Apache and haven't tried setting it up like this. 

However I noticed this in your config:
```
RewriteRule /theia/(.*)/     ws://localhost:3000/ [P,L]
```
Maybe you have to pass on whatever is matched via `(.*)` as well?
```
RewriteRule /theia/(.*)     ws://localhost:3000/$1 [P,L]
```
